plugins/de.fraunhofer.ipa.ros.xtext/src/de/fraunhofer/ipa/ros/validation/RosValidator.xtend

```diff
@@ -37,11 +42,83 @@ class RosValidator extends AbstractRosValidator {
 		if (Character.isUpperCase(c)){
 			error("The name of a package has to follow the ROS naming conventions: Capital letters are not allowed", null, INVALID_NAME);
 		}
- 	}}
+ 	}
+  }
+  
+  /* Customize Syntax Error Messages */
+  
+  /* For Topics */
+  @Check
+  def void CheckMsgsRefSubscriber(Node node){
```

Comment:
Sorry, this is not working as expected. It is marking as warning the full node. And so is quite hard to detect the line of the errors...

The checks have to be done (as my previous example) at interface level. That means:

```
@Check
def void CheckMsgsRefPublisher(Publisher pub){
     if (p.message.eContainer === null){
          warning("Couldn't resolve reference to TopicSpec \n" ... \n",
            null,INVALID_NAME)
      }
  	}
```